plutus-ir/test/TestLib.hs

```diff
@@ -1,89 +1,59 @@
+{-# LANGUAGE FlexibleContexts  #-}
 {-# LANGUAGE OverloadedStrings #-}
 module TestLib where
 
 import           Common
+import           PlcTestUtils
+import           PlutusPrelude                hiding ((</>))
 
+import           Control.Exception
+import           Control.Monad.Except
+import           Control.Monad.Reader         as Reader
+
+import qualified Language.PlutusCore.DeBruijn as PLC
+import           Language.PlutusCore.Pretty
 import           Language.PlutusCore.Quote
 import           Language.PlutusIR
+import           Language.PlutusIR.Parser     as Parser
+
+import           System.FilePath              (joinPath, (</>))
+
+import           Text.Megaparsec.Error        as Megaparsec
 
-import           Data.Text.Prettyprint.Doc
+import qualified Data.Text                    as T
+import qualified Data.Text.IO                 as T
 
-goldenPir :: String -> Term TyName Name a -> TestNested
-goldenPir name value = nestedGoldenVsDoc name $ pretty value
+withGoldenFileM :: String -> (T.Text -> IO T.Text) -> TestNested
+withGoldenFileM name op = do
+    dir <- currentDir
+    let testFile = dir </> name
+        goldenFile = dir </> name ++ ".golden"
+    return $ goldenVsTextM name goldenFile (op =<< T.readFile testFile)
+    where currentDir = joinPath <$> ask
 
-maybeDatatype :: Quote (Datatype TyName Name ())
-maybeDatatype = do
-    m <- freshTyName () "Maybe"
-    a <- freshTyName () "a"
-    match <- freshName () "match_Maybe"
-    nothing <- freshName () "Nothing"
-    just <- freshName () "Just"
+goldenPir :: Pretty b => (a -> b) -> Parser a -> String -> TestNested
```

Comment:
My brain does not want to process this series of utility functions but I think they're probably fine :dizzy_face: